Include/pymacro.h

```diff
@@ -122,8 +122,6 @@
 #  define Py_UNREACHABLE() __builtin_unreachable()
 #elif defined(__clang__) || defined(__INTEL_COMPILER)
 #  define Py_UNREACHABLE() __builtin_unreachable()
-#elif defined(_MSC_VER)
-#  define Py_UNREACHABLE() __assume(0)
```

Comment:
Is this change related? If it's only an issue in ceval.c, maybe redefine the macro there?

Reply:
As a fix only for x64, MSVC can use default case (`unknown opcode`) instead of current `TARGET(CACHE)` branch.

Reply:
`__assume(0)` in `TARGET(CACHE)` seems not to cause slowdown with this PR on x64. I'm fine with leaving as it is to avoid adding a confusing comment.

Reply:
Yes, this was related -- @neonene [claims ](https://github.com/faster-cpython/ideas/issues/321#issuecomment-1075315759)that code using `__assume(0)` performs worse than the default (see other thread). But I'm not convinced that it is worth it and his latest comments seem to indicate that it's only a problem on x86 (i.e., 32-bit), not on x64, so I think I'll take it out.